packages/firestore/test/unit/remote/bloom_filter.test.ts

```diff
@@ -0,0 +1,248 @@
+      expect.fail();
+    } catch (error) {
+      expect(
+        (error as Error)?.message.includes(
+          'INTERNAL ASSERTION FAILED: Invalid hash count: 0'
+        )
+      ).to.be.true;
+    }
+  });
+
+  it('mightContain in empty bloom filter should always return false', () => {
+    const bloomFilter = new BloomFilter(new Uint8Array(0), 0, 0);
+    expect(bloomFilter.mightContain('abc')).to.be.false;
+    expect(bloomFilter.mightContain('def')).to.be.false;
+  });
+
+  it('mightContain should always return false for empty string', () => {
+    const emptyBloomFilter = new BloomFilter(new Uint8Array(0), 0, 0);
+    const nonEmptyBloomFilter = new BloomFilter(
+      new Uint8Array([70, 204, 25]),
+      1,
+      16
+    );
+    expect(emptyBloomFilter.mightContain('')).to.be.false;
+    expect(nonEmptyBloomFilter.mightContain('')).to.be.false;
+  });
+
+  /**
+   * Golden tests are generated by backend based on inserting n number of
+   * documents path into a bloom filter.
+   *
+   * Full document path is generated by concating documentPrefix and number n,
+   * ie, projects/project-1/databases/database-1/documents/coll/doc12
+   *
+   * The test result is generated by checking the membership of documents from
+   * documentPrefix+0 to documentPrefix+2n. The membership results from 0 to n
+   * is expected to be true (1), and the membership results from n to 2n is
+   * expected to be false (0) with some false positive results (1).
+   *
+   * The test result of BloomFilter.mightContain() should match the backend
+   * result exactly.
+   */
+  describe('BloomFilter membership test', () => {
+    const documentPrefix =
+      'projects/project-1/databases/database-1/documents/coll/doc';
+
+    interface TestDataType {
+      bits: {
+        bitmap?: string;
+        padding?: number;
+      };
+      hashCount?: number;
+    }
+
+    interface TestResultType {
+      membershipTestResults: string;
+    }
+
+    function decodeBase64ToUint8Array(encoded: string): Uint8Array {
+      return ByteString.fromBase64String(encoded).toUint8Array();
+    }
+
+    function testBloomFilterAgainstExpectedResult(
+      bloomFilterInputs: TestDataType,
+      expectedResult: TestResultType
+    ): void {
+      const { bits, hashCount } = bloomFilterInputs;
+      const { membershipTestResults } = expectedResult;
+
+      const bloomFilter = new BloomFilter(
+        decodeBase64ToUint8Array(bits.bitmap || ''),
+        bits.padding || 0,
+        hashCount || 0
+      );
+      for (let i = 0; i < membershipTestResults.length; i++) {
+        const backendMembershipResult =
+          membershipTestResults[i] === '1' ? true : false;
+        const mightContain = bloomFilter.mightContain(documentPrefix + i);
+        expect(mightContain).to.equal(backendMembershipResult);
+      }
+    }
+
+    it('mightContain result for 1 document with 1 false positive rate', () => {
+      const testData = TEST_DATA.testDataCount1Rate1;
```

Comment:
Please remove the local variables `testData` and `testResult` and just call `testBloomFilterAgainstExpectedResult()` directly with the values from TEST_DATA. Removing unnecessary local variables like this often improves readability because there are fewer variables for human brains to keep track of.